### DISCUSSION TITLE
Re-invention Ushahidi as Portable, Lightweight, Modern one

### DISCUSSION THREAD

**Discussion by yuiseki** [UNopenGIS/7#18] [Ideas]
Ushahidi is one of the most well-known disaster information gathering GIS.
Ushahidi is OSS, however seems to be huge and overly complicated...

- https://github.com/ushahidi/platform
- https://github.com/ushahidi/platform-client

Disaster information gathering GIS that can be easily deployed in resource-limited environments is needed.

- If it already exists, I'd like to know.
- If we don't have it yet, why don't we make it?

**Comment by hfu** [comment]
I love your idea about a portable, lightweight, and modern version of Ushahidi for the 2020s. 

What came to my mind is something as easy as https://strapi.io/ in some Jamstack way ;-) 

Data collection is of course an important part of the information life cycle. As I see, one of the challenges is how we conduct proper information management (IM). Making the system portable and lightweight would solve some of the IM challenges. 

This idea would be something we can jointly elaborate on. We can share some of the target use cases and the problems to be solved by the new solution.

**Comment by yuiseki** [comment]
このテーマについて、関連技術を調査をしたため、簡単に報告します。日本語で失礼します。


## 調査の目的

- 資源の限られた環境下での地理空間情報の収集と発信のためのソフトウェアの技術的検討
- 条件
  - 資源の限られた環境下でも容易にデプロイ可能であること
  - オープンソースライセンスであること


## 調査の結論

- 第一の手段として、`openstreetmap-website` をRaspberry Pi上にデプロイし、スタンドアローンで動作可能にできないか検証したい
  - `openstreetmap-website` は、シンプルなシステム構成で、依存関係も少なく、さらにiDという優れた編集UIまで用意され、洗練されている
  - インターネットから遮断された環境でも `openstreetmap-website` が使えれば、マッパーたちは普段通りにその経験を活かせる
- 第二の手段として、StrapiでPostGISを前提とするプラグインを利用することを検討している
- 第三の手段として、Supabaseを使った最小のGIS CMSを作成することを検討している
- 最後の手段として、PostGISでもSpatiaLiteでも動くようなheadless CMS自体を独自で開発できないか検討している


## 調査の経緯

- 最初に、headless CMSであるStrapiを検討した
  - Strapiは非常にコンパクトで、使い心地もよかった
  - しかし、Strapiにはデフォルトでは地理空間情報を扱うためのフィールドがない
  - Strapiで地理空間情報を扱うためには、PostGISを前提とするサードパーティのプラグインを使う必要がある
- ここで、PostGISの運用を前提とするか否かがポイントとなりそうだと考えた
- PostGISよりも軽量なGIS DBである、SpatiaLiteの現状を調べた
  - SpatiaLiteはポータビリティは極めて高そうだが、ほぼ一人の人物によって開発が進められている
  - バインディングライブラリの選択肢も少なく、活用事例も少ない印象を受けた
  - SpatiaLiteの調査検討はここで中断した
- Raspberry Pi上でのPostGISの運用を前提とする方向で、Strapi以外のアプローチを検討した
  - OSSのBaaSであるSupabaseは、PostGISが最初から有効になっている、公式でPostGISのプラグインがある、などサポートが充実しており、活用事例も多いので、さらなる検討に値するかもしれない
  - 調べていくうちに気がついたが、PostGISの運用を前提とするのであれば、OpenStreetMap本家のRuby on Railsプロジェクトである `openstreetmap-website` を、そのままRaspberry Pi上にデプロイ可能かもしれない
    - 試してみたらデプロイ（ビルドおよび起動、動作確認）はできた
    - インターネットと通信している箇所にパッチを当てれば、スタンドアローンでも動かせるようになるかもしれない


## 調査の詳細

### Strapi

- 種別
  - headless CMS
- Webサイト
  - https://strapi.io/
- リポジトリ
  - https://github.com/strapi/strapi
- ライセンス
  - MIT Expat License
- GIS対応
  - 下記プラグインでPostGISに対応可能
    - https://github.com/am2222/strapi-plugin-postgis
- 感想
  - 管理画面の使い心地は良かった
  - PostGISの運用を前提とし、サードパーティプラグインに頼る必要がある


### Supabase

- 種別
  - Backend As A Service
- Webサイト
  - https://supabase.com/
- リポジトリ
  - https://github.com/supabase/supabase
- ライセンス
  - Apache License 2.0
- GIS対応
  - `postgis` という公式プラグインでPostGISに対応可能
    - https://www.supabase.jp/docs/guides/database/extensions
- 感想
  - 公式でPostGISに対応している
  - 管理画面の使い心地は微妙…


### openstreetmap-website

- 種別
  - Webサービス
- Webサイト
  - https://www.openstreetmap.org/
- リポジトリ
  - https://github.com/openstreetmap/openstreetmap-website
- ライセンス
  - GNU General Public License v2.0
- GIS対応
  - PostGISによってGISに対応している
- 感想
  - Ubuntu 22.04では問題なくdocker compose upできた
  - これをRaspberry Piで動くようにしてしまうのが近道かもしれない

## 最後に

PostGISをベースとしたheadless CMSに関する情報をお持ちの方がいましたら、ぜひ教えてください！
